src/core/lib/event_engine/posix_engine/ev_epoll1_linux.cc

```diff
@@ -375,7 +375,10 @@ void Epoll1Poller::Shutdown() {
 }
 
 void Epoll1Poller::Close() {
-  if (closed_) return;
+  {
+    grpc_core::MutexLock lock(&mu_);
+    if (closed_) return;
+  }
 
   if (g_epoll_set_.epfd >= 0) {
     close(g_epoll_set_.epfd);
```

Comment:
nit: I wonder if this needs to be done without holding the mutex. If not, it may be cleaner to take the lock once at the beginning of the function and hold it throughout.